src/transformers/modeling_utils.py

```diff
@@ -2276,11 +2286,25 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
                         )
                 dtype_orig = cls._set_default_torch_dtype(torch_dtype)
 
+            # Check if `_keep_in_fp32_modules` is not None
+            # but not force users to have `accelerate`
+            if cls._keep_in_fp32_modules is not None:
+                if not is_accelerate_available() and torch_dtype == torch.float16:
+                    use_keep_in_fp32_modules = False
+                    logger.warning(
+                        " `_keep_in_fp32_modules` is not set to `None` and you don't have `accelerate` installed"
+                        " it is recommended to have `accelerate` installed in this case `pip install accelerate`.",
```

Comment:
The warning is not really useful for a user that will just use `from_pretrained`. I'd say something like:
"For stability purposes, it is recommended to have `accelerate` installed when using this model in `torch.flaot16`, please install it with `pip install accelerate`."

Reply:
Thanks for the suggestion! Fixed in 1e80f14